App.tsx

```diff
@@ -27,11 +22,20 @@ const App = () => {
   };
 
   return (
-    <SafeAreaView style={backgroundStyle}>
-      <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
-      {/*<ProductListScreen />*/}
-        <ProductDetailsScreen />
-    </SafeAreaView>
+      <ProductsProvider>
+      <SafeAreaView style={backgroundStyle}>
+        <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
+          <NavigationContainer>
+              <Navigator initialRouteName="Home" screenOptions={{headerShown: false}}>
+                  <Group>
+                      <Screen name="Home" component={Drawer} />
```

Comment:
nesting Drawer in Stack - very debatable approach. I would suggest having StackNavigator inside DrawerNavigator.